### ISSUE TITLE
[BUG] Error while using DatePicker on edit forms.

### ISSUE THREAD

**Issue by turgayozgur** [pankod/refine#1259]
**Describe the bug**
Error while using DatePicker on edit forms. Seems like there is something going wrong with dayjs and antd's DatePicker which is default for refine.

```
dayjs.js:151 
Uncaught TypeError: date.locale is not a function
    at Object.format (dayjs.js:151)
    at formatValue (dateUtil.js:126)
    at useValueTexts.js:19
    at useMemo (useMemo.js:6)
    at useValueTexts (useValueTexts.js:8)
    at InnerPicker (Picker.js:135)
    at renderWithHooks (react-dom.development.js:14985)
```

**To Reproduce**
1. Add a new DatePicker to the edit form.
2. Nagivate to the edit page.
4. See error on the browser's console.

**Expected behavior**
No error.

**Screenshots**
![image](https://user-images.githubusercontent.com/2786640/142704656-edf5e3ca-4115-44c2-907b-b5ef22f6335c.png)

**Desktop (please complete the following information):**
 - OS: MacOS Monterey
 - Browser [Brawe]
 - Version 1.31.91 Chromium: 95.0.4638.69
 - Used dataProvider [custom]

**Comment by omeraplak** [comment]
Hey @turgayozgur ,
You can check our DatePicker example,

https://refine.dev/docs/examples/inputs/datePicker/

**Comment by turgayozgur** [comment]
Did you try to change the date on edit posts page? There are different types of errors while using datepicker on edit pages. I was got this error as well.
 
![F1452E1C-B662-4B44-9F7E-E62C6CF4195C](https://user-images.githubusercontent.com/2786640/142716693-e9c78dd0-9dff-4bbb-b337-5ae8729169b8.jpeg)

**Comment by omeraplak** [comment]
Hey @turgayozgur,
I found a similar issue for Antd DatePicker. https://github.com/ant-design/ant-design/issues/26190#issuecomment-703673400

I temporarily updated CodeSandbox -> https://codesandbox.io/s/refine-date-picker-example-r86xo?file=/src/App.tsx

We will investigate for a better solution Everything seems to be working for now.

Thank you for your attention!

**Comment by turgayozgur** [comment]
I got this error when I try to extend dayjs. Any ideas?

```
./prjectname/node_modules/@pankod/refine/src/components/antd/calendar.tsx
TypeScript error in ./prjectname/node_modules/@pankod/refine/src/components/antd/calendar.tsx(6,42):
Argument of type 'import("/Users/turgayozgur/Work/shopntier/cms-ui/node_modules/rc-picker/lib/generate/index").GenerateConfig<import("/Users/turgayozgur/Work/shopntier/cms-ui/node_modules/dayjs/index").Dayjs>' 
is not assignable to parameter of type 'import("./prjectname/node_modules/rc-picker/lib/generate/index").GenerateConfig<import("./prjectname/node_modules/@pankod/refine/node_modules/dayjs/index").Dayjs>'.  TS2345

    4 | // import "antd/es/calendar/style/index.css";
    5 | 
  > 6 | const Calendar = generateCalendar<Dayjs>(dayjsGenerateConfig);
      |                                          ^
    7 | 
    8 | export default Calendar;
    9 |

```

**Comment by omeraplak** [comment]
Could you try dayjs@1.10.5?

Like this; https://github.com/pankod/refine/blob/master/packages/core/package.json#L54

**Comment by turgayozgur** [comment]
```bash
yarn remove dayjs
yarn add dayjs@1.10.5
yarn dev
```
and still same.

**Comment by omeraplak** [comment]
I have this problem with yarn as far as I know. Can you try with npm instead of yarn? We'll release a patch for it next week.

**Comment by turgayozgur** [comment]
@omeraplak Thanks. It worked.

1. Use `npm` insted of `yarn`.
2. Use the same version that `refine` using for `dayjs`. It is currently `1.10.5`
3. Don't forget to wrap the date string value fetched from the API on `getValueProps`. Check it out the edit page on the example of [datepicker](https://codesandbox.io/s/refine-date-picker-example-r86xo?file=/src/App.tsx).

**Comment by omeraplak** [comment]
Hey @turgayozgur ,
We've released 2.2.2. Could you approve this issue is fixed?

**Comment by luizgribeiro** [comment]
Just tested it out on release 2.2.3 through yarn and it worked
